sdk/src/azure/iot/az_iot_common.c

```diff
@@ -27,6 +27,7 @@ AZ_NODISCARD az_result az_iot_message_properties_init(
   _az_PRECONDITION_NOT_NULL(properties);
   _az_PRECONDITION_VALID_SPAN(buffer, 0, true);
   _az_PRECONDITION(written_length >= 0);
+  _az_PRECONDITION(written_length <= az_span_size(buffer));
```

Comment:
Similar to https://github.com/Azure/azure-sdk-for-c/pull/1640, should this be considered a bug fix for the CL?